### PR TITLE
Bash Track, Grains, Add mentoring notes.

### PR DESCRIPTION
**Pull request by rpalo**
There aren't any mentoring notes for Bash!  So I stole a template from
the excellent Ruby track mentoring notes and created the first one.
I see a lot of the students run up against the same problems on Grains,
and there's a really slick pure Bash solution that most students are
really happy with.

### CODE REVIEW

**Comment by rpottsoh** [comment]
Someone from the @exercism/bash track should probably review and merge this.

File: tracks/bash/exercises/grains/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Grains
+Hello World -> **Grains** -> Armstrong Numbers -> Hamming -> ...
```
Comment (iHiD) [inline tracks/bash/exercises/grains/mentoring.md]: Could you remove this line please? It's something we will automatically inject into the website at a later way and I don't want the technical debt of it being included in text documents.

File: tracks/bash/exercises/grains/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ - Dollar signs aren't needed inside arithmetic substitutions.
+ - A lot of students us a `for` or `while` loop to calculate the total.  This is
+   slow, *especially* if they are sending out to `bc` to calculate each square.
+   This is a good time to suggest coming up with a solution that only performs
+   one calculation: either building a gigantic string to send to `bc`, or
+   calculating the total using a mathematic relationship (e.g. powers of 2).
+
+## Talking points
+
+ - If the student doesn't have a strong grasp on math, it seems effective to
```
Comment (iHiD) [inline tracks/bash/exercises/grains/mentoring.md]: You could probably drop the `-` here as it'll just make it a little less readable on the website.

Comment (rpalo) [reply tracks/bash/exercises/grains/mentoring.md]: Done!  I wasn't sure if you wanted to drop the bullet-point dashes in the previous section as well, so I didn't, but I'm happy to if it renders better on the site.

**Comment by iHiD** [comment]
I'm good with this from a website POV. I have no bash to know whether it's good or not, but I'm happy for it to get merged down and then refined later if necessary?

**Comment by kotp** [comment]
I would merge this, as it is, but no rights to do so.

**Comment by kotp** [comment]
Thanks @iHiD

**Pull request merged**